### ISSUE TITLE
comparing timescaledb vs InfluxDB vs redis

### ISSUE THREAD

**Issue by enthusiastics** [influxdata/influxdb#22342]
I'm looking into building some low latency real time application (<100ms) based on time series data. I'm comparing timescaledb vs InfluxDB vs Redis. I'm comparing them using https://db-engines.com/en/

From a lot of blog posts, I found that TimescaleDB had better performance than InfluxDB, but I found InfluxDB having in memory capabilities (while TimescaleDB does not). Then I wonder if by utilizing such in memory capabilities, would influx db perform better on timescaledb during live production.

Some industry expert mentioned Redis for such application. But it seems it is not really a database.

In summary my question is,

1, is TimescaleDB really not having in memory capabilities?

2, if I trigger in memory capabilities in InfluxDB, would the production level performance be probably better than TimescaleDB?

3, Is Redis a timeseries database, offering similar capabilities of TimescaleDB or InfluxDB?

**Comment by wbaker85** [comment]
Are you referring to the ability to run the metadata store in-memory, as documented here? https://docs.influxdata.com/influxdb/v2.0/reference/config-options/#store

That option allows you to run only the metadata store in memory, for InfluxDB 2.0. On 1.x InfluxDB, you can use (and it is the default) an "in-memory" index, see https://docs.influxdata.com/influxdb/v1.8/administration/config/#index-version--inmem - InfluxDB 2.0 only uses the TSI index.

I would not anticipate much difference in performance when running the metadata store in memory if you are using InfluxDB 2.0.

I can't comment on the specific capabilities of TimescaleDB with regards to running in-memory or not.

Redis is a key-value store. It's very good at storing and retrieving key and value data. There are unique characteristics of time-series data that InfluxDB is built specifically to address - [this article](https://www.influxdata.com/time-series-database/) & the links at the bottom might be helpful to check out to see some of those specifics.
